providers/unleash/pkg/provider_test.go

```diff
@@ -258,5 +298,6 @@ func TestMain(m *testing.M) {
 
 	cleanup()
 
+	os.Remove(backupFile)
 	os.Exit(exitCode)
```

Comment:
Maybe do it with defer?

Reply:
I tried defer, but it doesn't work because of `os.Exit(exitCode)'.

Reply:
i think this makes sense - only other option is to put both in a single defer func but i dont see much value in this